components/ai_chat/resources/untrusted_conversation_frame/components/conversation_entries/index.tsx

```diff
@@ -91,6 +91,12 @@ function ConversationEntries() {
           const lastEditedTime = latestTurn.createdTime
           const hasReasoning = latestTurnText.includes('<think>')
 
+          const turnModelKey = turn.modelKey
+            ? conversationContext.allModels
+                .find(m => m.key === turn.modelKey)
+                ?.key ?? undefined
+            : undefined;
```

Comment:
This is basically just confirming that the `modelKey` points to a valid model, right? Could we do this on the backend?

Reply:
turn.modelKey is what exactly is stored in the turn though, it could be null or a key that's no longer pointing to a valid model, so not super sure what should be pushed into backend.